lib/rucio/core/identity.py

```diff
@@ -47,7 +49,12 @@ def add_identity(identity, type, email, password=None, session=None):
 
     if type == IdentityType.USERPASS and password is not None:
         salt = os.urandom(255)  # make sure the salt has the length of the hash
-        password = hashlib.sha256('%s%s' % (salt, password.encode('utf-8'))).hexdigest()  # hash it
+        if six.PY3:
+            salt = salt.decode()
+            salted_password = ('%s%s' % (salt, password)).encode()
```

Comment:
i'm still not so super-happy about the whole `.encode()` business all over the codebase. is there a sensible way to do it more centrally? it's easy to forget to do it

Reply:
I think there is no easier way to do it, unfortunately